tmt/steps/discover/fmf.py

```diff
@@ -558,11 +559,42 @@ def assert_git_url(plan_name: Optional[str] = None) -> None:
             links=link_needles,
             excludes=excludes)
 
+        # Save fmf metadata
+        clonedir = self.clone_dirpath / 'tests'
+        for path in tmt.utils.filter_paths(self.testdir, ['.fmf']):
+            tmt.utils.copytree(path, clonedir / path.relative_to(self.testdir), dirs_exist_ok=True)
+
+        # Save upgrade plan
+        upgrade_path = self.get('upgrade_path')
+        if upgrade_path:
+            upgrade_path = f"{upgrade_path.lstrip('/')}.fmf"
+            (clonedir / upgrade_path).parent.mkdir()
+            shutil.copyfile(self.testdir / upgrade_path, clonedir / upgrade_path)
+            shutil.copymode(self.testdir / upgrade_path, clonedir / upgrade_path)
+
         # Prefix tests and handle library requires
         for test in self._tests:
             # Propagate `where` key
             test.where = cast(tmt.steps.discover.DiscoverStepData, self.data).where
 
+            # Save current test data
+            assert test.path is not None  # narrow type
+            relative_test_path = test.path.unrooted()
+            tmt.utils.copytree(
+                self.testdir / relative_test_path,
+                clonedir / relative_test_path,
+                dirs_exist_ok=True)
+            # Copy all parent main.fmf files
+            parent_dir = relative_test_path
+            while True:
+                parent_dir = parent_dir.parent
+                if (self.testdir / parent_dir / 'main.fmf').exists():
+                    shutil.copyfile(
+                        self.testdir / parent_dir / 'main.fmf',
+                        clonedir / parent_dir / 'main.fmf')
+                if parent_dir == Path('.'):
```

Comment:
The safest way of comparing two `Path` instances, avoiding pitfalls of symlinks and Python object ID comparison, is using `resolve()`. Plus there's `cwd()` method which is easier to track than `'.'`: `if parent_dir.resolve() == Path.cwd().resolve()`.

Reply:
Also, can this condition be tested directly by `while`?

```python
while parent_dir.resolve() != Path.cwd.resolve():
```

No way we could remove it by mistake, and it's immediately clear we're climbing a tree of paths.

Reply:
I'll use the cwd and resolve, but I can't really add it to while check as I need do..while loop for this, otherwise I would have to duplicate the check and copy before this loop